features/XouBaseApiLoginAndRegister/Service/XouBaseApiCalls.swift

```diff
@@ -0,0 +1,49 @@
+//
+//  XouBaseApiCalls.swift
+//  XouDevSpec
+//
+//  Created by Xander Schoeman on 2020/04/19.
+//
+
+import Foundation
+
+
+public struct XouBaseApiCalls {
+
+    public init() {
+    }
+    
+    public  func getUsers(completetionHandler: @escaping(Result<[User], AnimeError>) -> Void) {
+        
+        
+            guard let url = URL(string: "https://xoubase.herokuapp.com/users") else {
+                return
+            }
+        
+            var request = URLRequest(url: url,
+                cachePolicy: .useProtocolCachePolicy,
+            timeoutInterval: 10.0)
+
+
+        request.httpMethod = "GET"
+
+        let session = URLSession.shared
+        let dataTask = session.dataTask(with: request as URLRequest) { data, _, _ in
+            guard let jsonData = data else {
+                completetionHandler(.failure(.noDataAvailable))
+                return
+            }
+            do {
+                let decoder = JSONDecoder()
+                let response = try decoder.decode([User].self, from: jsonData)
+                let responseDetails = response
```

Comment:
responseDetails looks like the same as reponse ? Why the need for re-assigning it?

Reply:
Good Question... I'll give it a look while I make the other changes.

Reply:
So I need to specify the type which it decodes into, I cannot remove it as it requires a type. Do you have a suggestion as to how I can get around it or should I just keep it as is then?

Reply:
You can either remove "responseDetails = response" and then rename "response" to "responseDetails". Or you can still remove that line and pass "response" to the completionHandler instead

Reply:
Thank you for clearing that up. Done and done.

Reply:
I think what Zaheer is saying is that you can return the response variable in the completionHandler instead of reassigning it to another variable as it does not change anything. You can also try to specify the type of the variable by 

```
let response: [User] = ...
```

Reply:
So I tried changing it to inherit from : [User]
But I still asks me for a type when I try to decode it as it cannot go without a type. The decode function also does not have any overrides where there are no type arguments requested. I also tried returning the response variable in the completion handler but I am facing the same problem with the decoder.